### PR TITLE
feat(aaveOracle): override aave SC to extends functionallity

### PR DESCRIPTION
**Pull request by juandahl**
# feature: Add Redstone Aggregator with Chainlink Compatibility

## Description
This PR integrates the **Redstone Aggregator** into the system, allowing it to be configured via Bob configuration file. Additionally, a custom smart contract has been implemented to adapt the Redstone aggregator interface to be compatible with Chainlink's interface.

## Changes
- **Added Redstone Aggregator**: Integrated the Redstone price aggregator into the system for more flexible price feeds.
- **Bob Configuration**: Redstone aggregator can now be configured directly through Bob configuration files.
- **Custom Smart Contract**: Created a custom smart contract to adapt the Redstone aggregator interface to the Chainlink interface, ensuring compatibility with existing Chainlink-based setups.
- **Improved Price Feeds**: Enables the system to fetch and use prices from Redstone while maintaining compatibility with Chainlink's price infrastructure.

## Tests
- Verified Redstone aggregator is correctly integrated and configurable via Bob.
- Tested the custom smart contract to ensure proper interface adaptation between Redstone and Chainlink.
- Checked that the system can seamlessly use Redstone price feeds where required.

### CODE REVIEW

**Review by bit90s — APPROVED** [review]

File: contracts/dependencies/redstone/AggregatorRedStone.sol [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  IAggregatorRedstoneInterface public immutable oracle;
+  address public immutable asset;
+
+  constructor(address _oracle, address _asset) {
+    oracle = IAggregatorRedstoneInterface(_oracle);
+    asset = _asset;
+  }
+
+  function latestAnswer() external view returns (int256) {
+    return int256(oracle.priceOf(asset));
```
Comment (matzapata) [inline contracts/dependencies/redstone/AggregatorRedStone.sol:34]: Can a price be negative? if not let's use `uint256` instead

File: contracts/dependencies/redstone/AggregatorRedStone.sol [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    return oracle.getLatestRoundId();
+  }
+
+  function getAnswer(uint256 roundId) external pure returns (int256) {
+    revert ("Not implemented");
+  }
+
+  function getTimestamp(uint256 roundId) external pure returns (uint256) {
+    revert ("Not implemented");
+  }
```
Comment (matzapata) [inline contracts/dependencies/redstone/AggregatorRedStone.sol:51]: It's okay for this not to be implemented?

File: helpers/market-config-helpers.ts [hunk 3]
```diff
@@ -395,3 +396,36 @@ export const isIncentivesEnabled = (poolConfig: ICommonConfiguration) => {
...
+      const assets = poolConfig.ReserveAssets?.[network] ?? {};
+
+      for (const [key, asset] of Object.entries(assets)) {
+        const res = await deploy(`${key}${TESTNET_PRICE_AGGR_PREFIX}`, {
+          from: deployer,
+          args: [
+            redstoneAggregatorAddress,
+            asset
+          ],
+          ...COMMON_DEPLOY_PARAMS,
```
Comment (matzapata) [inline helpers/market-config-helpers.ts:422]: Ideally place this in 417 to ensure from and args are always overwritten

File: helpers/market-config-helpers.ts [hunk 4]
```diff
@@ -395,3 +396,36 @@ export const isIncentivesEnabled = (poolConfig: ICommonConfiguration) => {
...
+export const deployRedstoneAggregators = async (
+  deploy: DeploymentsExtension["deploy"],
+  deployer: tEthereumAddress,
+  redstoneAggregatorAddress: tEthereumAddress,
+  poolConfig: ICommonConfiguration,
+  network: eNetwork
+): Promise<ITokenAddress> => {
+  const isLive = hre.config.networks[network].live;
+
+  const result = {} as ITokenAddress;
```
Comment (matzapata) [inline helpers/market-config-helpers.ts:409]: I think you can do `const result: ITokenAddress = {}` instead of casting

**Review by matzapata — APPROVED** [review]
LGTM, just some questions and improvement ideas

**Pull request merged**